src/helpers/dates.js

```diff
@@ -0,0 +1,90 @@
+/* eslint-disable no-nested-ternary */
+/* eslint-disable eqeqeq */
+/* eslint-disable no-extend-native */
+/* eslint-disable func-names */
+const isYear = (date) => date.length === 4;
+const isUnknown = (date) => (date == '99' ? '01' : date);
+const isDashed = (date) => date.includes('-');
+const splitAt = (index) => (x) => [x.slice(0, index), x.slice(index)];
+
+Array.prototype.lastElement = function () {
+  return this.length ? this[this.length - 1] : undefined;
+};
+
+// handles dates without formatting, withoutslashes or dashes
+const compactDate = (date, age = null) => {
+  const firstFour = date.substr(0, 4);
+  const lastFour = date.substr(date.length - 4);
+  const thisYear = new Date().getFullYear();
+  const year = Number(firstFour) > 1920 && Number(firstFour) <= thisYear
+    ? firstFour
+    : lastFour;
+
+  const dates = year === firstFour ? lastFour : firstFour;
+  const datesArr = splitAt(2)(dates).map((item) => item.replace('99', '01'));
+
+  const formatedDate = datesArr.lastElement > 12 ? datesArr.reverse() : datesArr;
+
+  const birthYear = age ? thisYear - Number(age) : year.replace('9999', '1970');
+
+  return `${birthYear}-${formatedDate.join('-')}`;
+};
+
+// handles dates with slashes or dashes
+const slashedDate = (date) => {
+  const splitDate = isDashed(date) ? date.split('-') : date.split('/');
+
+  const dateArr = splitDate.map((item) => (isYear(item) ? item : isUnknown(item)));
+
+  const year = dateArr.filter((item) => isYear(item)).join('');
+  const dates = dateArr.filter((item) => !isYear(item));
+  const formatedDate = dates.lastElement > 12 ? dates.reverse() : dates;
+
+  return `${year}-${formatedDate.join('-')}`;
+};
+
+const emptyBirthDate = (date, age) => {
+  if (date?.toString().length < 8) return '1970-01-01';
+  const year = age ? new Date().getFullYear() - Number(age) : '1970';
```

Comment:
The year should be computed based on the `reporting date`, not today's date, unless the reporting date is blank too.

Reply:
This was to handle the birthdates field but it has now been refactored and readable. Reporting date was basically a copy of formatted INCID